### PR TITLE
test: Add test for `sendmany` rpc that uses `subtractfeefrom` parameter

### PR DESCRIPTION
**Pull request by yusufsahinhamza**
This PR adds test that uses `sendmany` rpc to send **BTC** to multiple addresses using `subtractfeefrom` parameter, then checks receiver addresses balances to make sure fees are subtracted correctly.

### CODE REVIEW

File: test/functional/wallet_basic.py [hunk 1]
```diff
@@ -267,6 +267,20 @@ def run_test(self):
         assert_equal(self.nodes[2].getbalance(), node_2_bal)
         node_0_bal = self.check_fee_amount(self.nodes[0].getbalance(), node_0_bal + Decimal('10'), fee_per_byte, self.get_vsize(self.nodes[2].gettransaction(txid)['hex']))
 
+        # Sendmany 5 BTC to two address with subtracting fee from both
+        a0 = self.nodes[0].getnewaddress()
+        a1 = self.nodes[0].getnewaddress()
+        txid = self.nodes[2].sendmany('', {a0: 5, a1: 5}, 0, "", [a0, a1])
```
Comment (brunoerg) [inline test/functional/wallet_basic.py:273]: Feel free to ignore, just a simple suggestion to make it friendly to read:
```suggestion
        txid = self.nodes[2].sendmany(dummy='', amounts={a0: 5, a1: 5}, minconf=0, comment="", subtractfeefrom=[a0, a1])
```

Comment (yusufsahinhamza) [reply test/functional/wallet_basic.py:273]: @brunoerg I think that's better, thanks for the suggestion.

File: test/functional/wallet_basic.py [hunk 2]
```diff
@@ -267,6 +267,20 @@ def run_test(self):
         assert_equal(self.nodes[2].getbalance(), node_2_bal)
         node_0_bal = self.check_fee_amount(self.nodes[0].getbalance(), node_0_bal + Decimal('10'), fee_per_byte, self.get_vsize(self.nodes[2].gettransaction(txid)['hex']))
 
+        # Sendmany 5 BTC to two address with subtracting fee from both
```
[the remark this answers is not in the document]
Comment (yusufsahinhamza) [reply test/functional/wallet_basic.py:270]: @kiminuo Nice suggestion, i'm also open for better explanatory comment if you would like to tell any.

File: test/functional/wallet_basic.py [hunk 3]
```diff
@@ -267,6 +267,20 @@ def run_test(self):
...
+        a0 = self.nodes[0].getnewaddress()
+        a1 = self.nodes[0].getnewaddress()
+        txid = self.nodes[2].sendmany(dummy='', amounts={a0: 5, a1: 5}, minconf=0, comment='', subtractfeefrom=[a0, a1])
```
[the remark this answers is not in the document]
Comment (yusufsahinhamza) [reply test/functional/wallet_basic.py:273]: @kiminuo Other tests above and below also uses "comment" parameter but not with keyword/named arguments, i've also tried it with only necessary parameters and had no issue, but i wanted to stick to them anyway.

I'm not entirely sure if they're needed, and i was actually thinking about to remove them.

[the remark this answers is not in the document]
Comment (yusufsahinhamza) [reply test/functional/wallet_basic.py:273]: @kiminuo Please consider reading my previous answer again. I would be more appreciated for you to tell me whether you think it's necessary or not, as i told you that i'm not sure.

[the remark this answers is not in the document]
Comment (yusufsahinhamza) [reply test/functional/wallet_basic.py:273]: @kiminuo It seems like not necessary, i'll remove it in the next commit, thanks for the discussion.

[the remark this answers is not in the document]
Comment (yusufsahinhamza) [reply test/functional/wallet_basic.py:273]: @kiminuo Checking rpc docs of `sendmany`, **"comment"** parameter is optional and doesn't seem necessary in this test case, so i think i'll just remove it, thanks again.

File: test/functional/wallet_basic.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (yusufsahinhamza) [reply test/functional/wallet_basic.py:270]:
```suggestion
        # Sendmany 5 BTC to two addresses with subtracting fee from both addresses
```

A little addition, this seems better, thanks for the suggestion.

**Comment by achow101** [comment]
ACK 057057a2d7e23c2e29cbfd29a5124b3947a33757

**Pull request merged**